lib/fluent/plugin/in_forward.rb

```diff
@@ -269,69 +542,62 @@ def initialize(io, linger_timeout, log, on_message)
             remote_port = nil
             remote_addr = nil
           end
-          "accepted fluent socket from '#{remote_addr}:#{remote_port}': object_id=#{self.object_id}"
+          [ "accepted fluent socket", {address: remote_addr, port: remote_port, instance: self.object_id} ]
         }
       end
 
       def on_connect
+        @on_connect_callback.call(self)
       end
 
-      def on_read(data)
-        first = data[0]
-        if first == '{' || first == '['
-          m = method(:on_read_json)
-          @serializer = :to_json.to_proc
-          @y = Yajl::Parser.new
-          @y.on_parse_complete = lambda { |obj|
-            option = @on_message.call(obj, @chunk_counter, @peeraddr)
-            respond option
-            @chunk_counter = 0
-          }
-        else
-          m = method(:on_read_msgpack)
-          @serializer = :to_msgpack.to_proc
-          @u = Fluent::Engine.msgpack_factory.unpacker
+      # API to register callback for data arrival
+      def on_data(delimiter: nil, &callback)
```

Comment:
It looks code for input data formatted as JSONL (JSON lines delimited by newlines).
And I found that it's not implemented in this plugin... I'll check it again later.